### DISCUSSION TITLE
[Question] Is it possible to disable the MouseGestures script on a per website basis?

### DISCUSSION THREAD

**Discussion by TFWol** [xiaoxiaoflood/firefox-scripts#316] [General Support] [answered]
Ideally, I'd like to only disable specific gestures for some sites and not for others, but disabling the script entirely for certain websites is fine.

One reason is some of the LAN tools I use leverage their own gestures like right-click drag, but it causes problems with the script on.
Trying to use `// @exclude` didn't work so I've been manually disabling the script via your userChromeJS Manager menu, but it gets very monotonous.

**Comment by xiaoxiaoflood** [comment, marked as answer]
Hi, I don't intend on adding this feature, but it could be done here:

https://github.com/xiaoxiaoflood/firefox-scripts/blob/0bbf504dedd793790e94dc88f8f90f3f18429132/chrome/mouseGestures.uc.js#L313

By adding a few lines.  For instance:

```js
  stopGesture: async function (gst, win) {
    if (win.gBrowser.currentURI.spec.startsWith('https://example.com/'))
      return;
```

In this case, MGest won't work if current tab URL starts with `https://example.com/`.

**Reply by TFWol** [reply]
That did the trick, thanks so much!
You gave me enough to work with so I was able to leverage an LLM.

I ended up placing this:
```javascript
  // Urls where script should not apply 
  EXCLUDED_URL_PREFIXES: [
    'http://localhost:1234/',
    'https://example2.com/',
    // Add more URL prefixes here as needed
  ],
```
above this line to easily add more urls:
https://github.com/xiaoxiaoflood/firefox-scripts/blob/0bbf504dedd793790e94dc88f8f90f3f18429132/chrome/mouseGestures.uc.js#L24

and then this:
```javascript
    // Check if the current URL should be excluded
    if (this.EXCLUDED_URL_PREFIXES.some(prefix => win.gBrowser.currentURI.spec.startsWith(prefix))) {
      return;
    }
```

after that line you mentioned:
https://github.com/xiaoxiaoflood/firefox-scripts/blob/0bbf504dedd793790e94dc88f8f90f3f18429132/chrome/mouseGestures.uc.js#L313

**Comment by TFWol** [comment]
Is there a simple reference somewhere that walks through how to make these particular types of scripts which affect the browser itself? Or does this require intimately knowing how Firefox internals work?

I know what kinds of keywords to use for regular scripts (userscripts, greasemonkey, greasyfork, etc), but not for this stuff.

**Reply by xiaoxiaoflood** [reply]
- https://searchfox.org/
- https://firefox-source-docs.mozilla.org/

There was also the old MDN docs, available in Internet Archive. Much of the info there is still relevant, but obviously much have changed since then.

http://web.archive.org/web/20210206223326/https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Multiprocess_Firefox/Message_Manager/Message_manager_overview

You should play with [Browser Toolbox](https://firefox-source-docs.mozilla.org/devtools-user/browser_toolbox/index.html).

Reading code from legacy extensions and userChromeJS scripts also helps.

I don't know how long you're using Firefox, but you should know in the past (til 2017) extensions had unrestricted access to Firefox internals, just like every piece of code in this repo. Extensions I update here are just a few of them, the ones I find useful enough for me to keep updating them so they can still be used in current Firefox.

Scripts for userChromeJS are like regular UserScripts, but targeting Firefox browser instead of webpages. They have the same power as legacy extensions, but are more suited for less complex tasks and usually don't present an UI for the user to change settings, as almost always they are served as a single JS file.
